src/reducers/noteReducer.js

```diff
@@ -1,14 +1,72 @@
 export const noteReducer = (state, action) => {
-  const index = state.findIndex((el) => el.id === action.payload.id);
   switch (action.type) {
     case "ADD_NOTE":
-      return [...state, { ...action.payload, id: state.length + 1, createdOn: new Date() }];
-    case "EDIT_NOTE":
-      return [...state.slice(0, index), { ...state[index], ...action.payload }, ...state.slice(index + 1)];
-    case "ARCHIVE_NOTE":
-      return state.filter((el) => el.id !== action.payload.id);
-    case "DELETE_NOTE":
-      return state.filter((el) => el.id !== action.payload.id);
+      return {
+        ...state,
+        allNotes: [...state.allNotes, { ...action.payload, id: new Date().toLocaleString(), createdOn: new Date() }],
+      };
+    case "EDIT_NOTE": {
+      const allNotesIndex = state.allNotes.findIndex((el) => el.id === action.payload.id);
+      return {
+        ...state,
+        allNotes: [
+          ...state.allNotes.slice(0, allNotesIndex),
+          { ...state.allNotes[allNotesIndex], ...action.payload },
+          ...state.allNotes.slice(allNotesIndex + 1),
+        ],
```

Comment:
suggestion: you can try map function here.
this might work (not tried) -

allNotes : state.allNotes.map( (note) =>
    note.id === action.payload.id ? { ...note, ...action.payload } : note
);

Reply:
Oh this is an efficient and readable way of sending a new map array.. have integrated this. thanks!